src/index.rs

```diff
@@ -0,0 +1,119 @@
+#![allow(missing_docs)]
+use std::convert::TryFrom;
+
+bitfield::bitfield! {
+    /// An H3 index.
+    #[derive(Debug, Copy, Clone, PartialEq, Eq)]
+    #[cfg_attr(
+        feature = "serde-support",
+        derive(serde::Serialize, serde::Deserialize),
+        serde(transparent)
+    )]
+    pub struct Index(u64);
```

Comment:
Rename to `Cell` to better describe the fact that we constrain valid representations to only H3 cell mode.